client/components/data/query-post-stats/index.jsx

```diff
@@ -1,80 +1,43 @@
-import { isEqual } from 'lodash';
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
 import { requestPostStats } from 'calypso/state/stats/posts/actions';
 import { isRequestingPostStats } from 'calypso/state/stats/posts/selectors';
 
-class QueryPostStats extends Component {
-	static defaultProps = {
-		requestPostStats: () => {},
-		heartbeat: 0,
-	};
+function useMemoizedFields( fields ) {
+	const memoizedFields = useRef();
+	const joinedArray = fields.join();
 
-	static propTypes = {
-		siteId: PropTypes.number,
-		postId: PropTypes.number,
-		fields: PropTypes.array,
-		requestingPostStats: PropTypes.bool,
-		requestPostStats: PropTypes.func,
-		heartbeat: PropTypes.number,
-	};
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillMount() {
-		const { requestingPostStats, siteId, postId } = this.props;
-		if ( ! requestingPostStats && siteId && typeof postId !== 'undefined' ) {
-			this.requestPostStats( this.props );
-		}
-	}
-
-	componentWillUnmount() {
-		this.clearInterval();
+	if ( joinedArray !== memoizedFields.current ) {
+		memoizedFields.current = joinedArray;
```

Comment:
Here you're storing and returning the joined array (string) but you want to return the actual array.